lib/windows.cc

```diff
@@ -72,34 +79,35 @@ Napi::Object getActiveWindow(const Napi::CallbackInfo &info) {
 
 std::vector<Window> _windows;
 
-BOOL CALLBACK EnumWindowsProc(HWND hwnd, LPARAM lparam) {
+BOOL CALLBACK EnumWindowsProc(HWND hwnd, LPARAM lparam)
+{
     auto process = getWindowProcess(hwnd);
     _windows.push_back({process, reinterpret_cast<int64_t>(hwnd)});
     return TRUE;
 }
 
-Napi::Array getWindows(const Napi::CallbackInfo &info) {
+Napi::Array getWindows(const Napi::CallbackInfo &info)
+{
     Napi::Env env{info.Env()};
 
     _windows.clear();
     EnumWindows(&EnumWindowsProc, NULL);
 
-    auto arr = Napi::Array::New(env, _windows.size());
+    auto arr = Napi::Array::New(env);
+    for (auto _win : _windows)
+    {
 
-    for (int i = 0; i < _windows.size(); i++) {
         auto obj{Napi::Object::New(env)};
-
-        obj.Set("id", _windows[i].id);
-        obj.Set("processId", _windows[i].process.pid);
-        obj.Set("path", _windows[i].process.path);
-
-        arr[i] = obj;
+        obj.Set("id", _win.id);
+        obj.Set("processId", _win.process.pid);
+        obj.Set("path", _win.process.path);
+        arr.Set(_win.id, obj);
```

Comment:
Shouldn't it be `arr.Set(i, obj)`?

Reply:
I'd rather like to avoid empty items...
![image](https://user-images.githubusercontent.com/11065386/62006584-b4282b00-b142-11e9-9183-e876b30cd0ad.png)

Reply:
> Shouldn't it be `arr.Set(i, obj)`?

yes I could use that, but I prefer this code where I just have to use an existed property.

Reply:
> I'd rather like to avoid empty items...
> ![image](https://user-images.githubusercontent.com/11065386/62006584-b4282b00-b142-11e9-9183-e876b30cd0ad.png)

okay, I will do something about it